File: taller/JCZhang-Taller/JCZhang-Taller-Java/src/main/java/models/Persona.java
package models;

public abstract class Persona {


    public Persona(int id, String nombre, int aniosExp) {
        this.id = nextId();
        this.nombre = nombre;
        this.aniosExp = aniosExp;

    }
    int id;
    String nombre;
    int aniosExp;

    protected Persona() {
    }

    public int nextId() {
        int id = 0;
        return id++;
    }
    abstract void saludar ();
}

File: taller/MarioBueno_Taller_Kotlin/src/main/kotlin/models/Persona.kt
//Mario Bueno López
//maarioo2525@gmail.com

package models

abstract class Persona (var id: Int = nextId(), var nombre: String, var experiencia: Int){
    companion object {
        var id = 0
        fun nextId(): Int {
            return id++
        }
    }
    abstract fun saludar()
}